catalog/app/containers/Bucket/PackageRevisions.js

```diff
@@ -428,21 +432,56 @@ export default function PackageRevisions({
     }
   })
 
-  const revisionCountData = useRevisionCountData({ bucket, name })
+  const [key, setKey] = React.useState(1)
+
+  const revisionCountData = useRevisionCountData({ bucket, name, key })
   const revisionsData = useRevisionsData({
     bucket,
     name,
     page: actualPage,
     perPage: PER_PAGE,
+    key,
   })
-  const countsData = useCountsData({ bucket, name })
+  const countsData = useCountsData({ bucket, name, key })
+
+  const onExited = React.useCallback(
+    (res) => {
+      // refresh data if new revision of current package was pushed
+      if (res && res.pushed && res.pushed.name === name) {
+        setKey(R.inc)
+        return true
+      }
+      return false
+    },
+    [name, setKey],
+  )
+
+  const updateDialog = usePackageUpdateDialog({ bucket, name, onExited })
 
   return (
     <M.Box pb={{ xs: 0, sm: 5 }} mx={{ xs: -2, sm: 0 }}>
-      <M.Box pt={{ xs: 2, sm: 3 }} pb={{ xs: 2, sm: 1 }} px={{ xs: 2, sm: 0 }}>
+      {updateDialog.render()}
+
+      <M.Box
+        pt={{ xs: 2, sm: 3 }}
+        pb={{ xs: 2, sm: 1 }}
+        px={{ xs: 2, sm: 0 }}
+        display="flex"
+      >
         <M.Typography variant="h5" ref={scrollRef}>
-          {name}
+          <StyledLink to={urls.bucketPackageDetail(bucket, name)}>{name}</StyledLink>{' '}
+          revisions
         </M.Typography>
+        <M.Box flexGrow={1} />
+        <M.Button
```

Comment:
It could be standalone component `ButtonPushRevisions`, reused by `PackageTree` page

Reply:
well, not much to reuse here. also, PackageTree and PackageRevisions use differently size buttons